modules/user/src/main/ui/UserShowSide.scala

```diff
@@ -30,7 +30,7 @@ final class UserShowSide(helpers: Helpers):
           "active" -> active.contains(pk)
         ),
         href := ctx.pref.showRatings.so:
-          if isPuzzle then routes.Puzzle.dashboard(Days(30), "home", u.username.some).url
+          if isPuzzle && ctx.is(u) then routes.Puzzle.dashboard(Days(30), "home", u.username.some).url
           else routes.User.perfStat(u.username, pk).url
```

Comment:
`routes.User.perfStat(u.username, pk)` doesn't make any sense when `pk == puzzle`. I'll make sure it's a 404.

Reply:
@ornicar it goes to this page though: `https://lichess.org/@/ChessTheory64/perf/puzzle`

Reply:
Yes and it's non-sensical because that page is about games, not puzzles. I'm working on adding type safety to avoid it.